PAYECalc/TaxRate.cs

```diff
@@ -43,6 +48,12 @@ public class TaxRate
         /// <param name="chargeableIncome">Tax rate chargeable income</param>
         public TaxRate(double rate, decimal chargeableIncome)
         {
+            //reject if rate is < 0 
+            if (rate < 0) throw new ArgumentException(INVALID_TAXRATE_EXCEPTON_MESSAGE);
+
+            //reject if chargeAbleIncome is <= 0 
+            if (chargeableIncome <= 0) throw new ArgumentException(INVALID_CHARGEABLEINCOME_EXCEPTON_MESSAGE);
```

Comment:
There is no need for this, this is because setting the Rate and ChargeableIncome properties with invalid values will automatically throw the exceptions we have in place.

Reply:
Explain please. How about the client assessing a tax rate member after creation  and setting invalid values?
Should these be readonly fields then? Assigned only at construction?

Reply:
Our validations will reject it, a runtime exception will be thrown. Remember, we have exposed public properties but in the class we are using private fields which allows us to do our validations, so the set function in each of the properties will be called whenever new values are coming in, and when the validations fails, the exception will be thrown.